Content.Server/Replays/ReplayEventSystem.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System.Linq;
+using System.Text;
+using Content.Server.Mind;
+using Content.Server.Pinpointer;
+using Content.Server.Prayer;
+using Content.Shared.CCVar;
+using Content.Shared.MassMedia.Systems;
+using Content.Shared.Mobs;
+using Content.Shared.Roles;
+using Content.Shared.Slippery;
+using Content.Shared.Stunnable;
+using Robust.Server.GameObjects;
+using Robust.Shared.Configuration;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Robust.Shared.Replays;
+using Robust.Shared.Serialization.Manager;
+using Robust.Shared.Serialization.Markdown.Mapping;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Server.Replays;
+
+public sealed class ReplayEventSystem : EntitySystem
+{
+    [Dependency] private readonly IReplayRecordingManager _replays = default!;
+    [Dependency] private readonly ISerializationManager _serialman = default!;
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+    [Dependency] private readonly TransformSystem _transformSystem = default!;
+    [Dependency] private readonly NavMapSystem _navMapSystem = default!;
+    [Dependency] private readonly MindSystem _mindSystem = default!;
+    [Dependency] private readonly SharedRoleSystem _roles = default!;
+    [Dependency] private readonly IConfigurationManager _cfg = default!;
+    [Dependency] private readonly MapSystem _mapSystem = default!;
+
+    private List<ReplayEvent>? _replayEvents = new();
+    private bool _recordEvents;
+
+    public override void Initialize()
+    {
+        _replays.RecordingStopped2 += ReplaysOnRecordingStopped;
+        _replays.RecordingStarted += ReplaysOnRecordingStarted;
+
+        // Using an event here because mob stuff is in shared and we cannot call RecordReplayEvent from there.
```

Comment:
You should absolutely make a shared version of this system that just does nothing, to avoid fragmentation like this.

Reply:
How does one make a shared system that basically only exists server side?

Reply:
Make it have only virtual methods in shared that aren't overriden on the client, so they won't do anything.